File: repos_estudiantes_TARQUI_tercera_entrega/ILLANES_ESCOBAR_FERNANDO_JOHSUA/PERSISTENCIA/Ejercicio5/Java/Sucursal.java
package PERSISTENCIA.Ejercicio5.Java;

import java.io.Serializable;
import java.util.ArrayList;

public class Sucursal implements Serializable {
    private int numero;
    private String direccion;
    private ArrayList<Medicamento> medicamentos;

    public Sucursal(int numero, String direccion) {
        this.numero = numero;
        this.direccion = direccion;
        this.medicamentos = new ArrayList<>();
    }

    public void agregarMedicamento(Medicamento m) {
        medicamentos.add(m);
    }

    public int getNumero() { return numero; }
    public String getDireccion() { return direccion; }
    public ArrayList<Medicamento> getMedicamentos() { return medicamentos; }

    @Override
    public String toString() {
        return "Sucursal " + numero + " - " + direccion + ": " + medicamentos;
    }
}

File: repos_estudiantes_TARQUI_tercera_entrega/ILLANES_ESCOBAR_FERNANDO_JOHSUA/PERSISTENCIA/Ejercicio5/Python/sucursal.py
from medicamento import Medicamento

class Sucursal:
    def __init__(self, numero: int, direccion: str):
        self.numero = numero
        self.direccion = direccion
        self.medicamentos: list[Medicamento] = []

    def agregar_medicamento(self, medicamento: Medicamento):
        self.medicamentos.append(medicamento)

    def __str__(self):
        return f"Sucursal {self.numero} - {self.direccion}"
